CMake/itkExternal_Eigen3.cmake

```diff
@@ -36,33 +39,41 @@ else()
   # However, generators and c,cxx compilers have to be explicitly passed
   # for CMake configuration to work.
   # Configure Eigen
-  file(MAKE_DIRECTORY ${_eigen3_build_dir})
   execute_process(
     COMMAND
-      ${CMAKE_COMMAND} ${_eigen3_source_dir} "-DCMAKE_APPLE_SILICON_PROCESSOR=${CMAKE_APPLE_SILICON_PROCESSOR}"
-      "-DCMAKE_INSTALL_PREFIX=${_eigen3_cmake_install_prefix}"
-      "-DCMAKE_INSTALL_INCLUDEDIR=${_eigen3_cmake_install_includedir}"
-      "-DCMAKE_INSTALL_DATADIR=${_eigen3_cmake_install_datadir}" "-DCMAKE_GENERATOR=${CMAKE_GENERATOR}"
-      "-DCMAKE_SH=${CMAKE_SH}" "-DCMAKE_GENERATOR_TOOLSET=${CMAKE_GENERATOR_TOOLSET}" "-DCMAKE_SH=${CMAKE_SH}"
-      "-DCMAKE_GENERATOR_PLATFORM=${CMAKE_GENERATOR_PLATFORM}" "-DCMAKE_GENERATOR_INSTANCE=${CMAKE_GENERATOR_INSTANCE}"
-      "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}" "-DCMAKE_C_FLAGS=${CMAKE_C_FLAGS}"
-      "-DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}" "-DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}"
-      "-DCMAKE_OSX_ARCHITECTURES=${CMAKE_OSX_ARCHITECTURES}" "-DCMAKE_SYSTEM_VERSION=${CMAKE_SYSTEM_VERSION}"
+      ${CMAKE_COMMAND}
+      -DCMAKE_SYSTEM_VERSION=${CMAKE_SYSTEM_VERSION}
+      -DCMAKE_GENERATOR=${CMAKE_GENERATOR}
+      -DCMAKE_GENERATOR_TOOLSET=${CMAKE_GENERATOR_TOOLSET}
+      -DCMAKE_GENERATOR_PLATFORM=${CMAKE_GENERATOR_PLATFORM}
+      -DCMAKE_GENERATOR_INSTANCE=${CMAKE_GENERATOR_INSTANCE}
+      -DCMAKE_SH=${CMAKE_SH}
+      -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+      -DCMAKE_C_FLAGS=${CMAKE_C_FLAGS}
+      -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+      -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}
+      -DCMAKE_OSX_ARCHITECTURES=${CMAKE_OSX_ARCHITECTURES}
+      -DCMAKE_INSTALL_PREFIX=${_eigen3_cmake_install_prefix}
+      -DCMAKE_INSTALL_INCLUDEDIR=${_eigen3_cmake_install_includedir}
+      -DCMAKE_INSTALL_DATADIR=${_eigen3_cmake_install_datadir}
+      -DITK_USE_EIGEN_MPL2_ONLY:BOOL=${ITK_USE_EIGEN_MPL2_ONLY}
       ${_additional_external_project_args}
```

Comment:
```suggestion
        -DCMAKE_SYSTEM_VERSION:STRING=${CMAKE_SYSTEM_VERSION}
        -DCMAKE_GENERATOR:STRING=${CMAKE_GENERATOR}
        -DCMAKE_GENERATOR_TOOLSET:STRING=${CMAKE_GENERATOR_TOOLSET}
        -DCMAKE_GENERATOR_PLATFORM:STRING=${CMAKE_GENERATOR_PLATFORM}
        -DCMAKE_GENERATOR_INSTANCE:STRING=${CMAKE_GENERATOR_INSTANCE}
        -DCMAKE_SH:PATH=${CMAKE_SH}
        -DCMAKE_C_COMPILER:PATH=${CMAKE_C_COMPILER}
        -DCMAKE_C_FLAGS:STRING=${CMAKE_C_FLAGS}
        -DCMAKE_CXX_COMPILER:PATH=${CMAKE_CXX_COMPILER}
        -DCMAKE_CXX_FLAGS:STRING=${CMAKE_CXX_FLAGS}
        -DCMAKE_OSX_ARCHITECTURES:STRING=${CMAKE_OSX_ARCHITECTURES}
        -DCMAKE_INSTALL_PREFIX:PATH=${_eigen3_cmake_install_prefix}
        -DCMAKE_INSTALL_INCLUDEDIR:PATH=${_eigen3_cmake_install_includedir}
        -DCMAKE_INSTALL_DATADIR:PATH=${_eigen3_cmake_install_datadir}
        -DITK_USE_EIGEN_MPL2_ONLY:BOOL=${ITK_USE_EIGEN_MPL2_ONLY}
        ${_additional_external_project_args}
```

Reply:
To follow-up on https://github.com/InsightSoftwareConsortium/ITK/pull/4265#discussion_r1376841622:

> * look into the need for passing `CMAKE_SH`
> * consider passing `CMAKE_BUILD_TYPE:STRING=Release` in the case of single config generator

Reply:
Fixed the style, thanks!

Reply:
> re: not building anything

Since we are calling `${CMAKE_COMMAND} --build . --config Release --target install` further down, I suggest we also pass `CMAKE_BUILD_TYPE` if we are in the context of a single-configuration IDE.

```cmake
if(NOT CMAKE_CONFIGURATION_TYPES)
  list(APPEND _additional_external_project_args
    -DCMAKE_BUILD_TYPE:STRING=Release)
endif()
```

Reply:
> re: CMAKE_SH

In that case, consider adding the following:

```cmake
if(CMAKE_SH AND CMAKE_VERSION VERSION_LESS "3.17.0" AND )
  # Setting CMAKE_SH is required when using "MinGW Makefiles" generator with CMake < 3.17
  # See https://github.com/InsightSoftwareConsortium/ITK/issues/66#issuecomment-424374973
  list(APPEND _additional_external_project_args
    -DCMAKE_SH:PATH=${CMAKE_SH})
endif()
```

Reply:
Thanks @jcfr , added both suggestions.